Source/AwsCommonRuntimeKit/mqtt/MqttRequestResponseClient.swift

```diff
@@ -181,60 +220,154 @@ public struct RequestResponseOperationOptions: CStructWithUserData, Sendable {
+            options.topic_filter = topicFilterCursor
+            options.user_data = userData
+            return body(options)
+        }
     }
-    
 }
 
- /// A streaming operation is automatically closed (and an MQTT unsubscribe triggered) when its
- /// destructor is invoked.
-public class StreamingOperation {
+// IMPORTANT: You are responsible for concurrency correctness of StreamingOperationCore.
+// The rawValue is mutable cross threads and protected by the rwlock.
+/// The internal core of the streaming operation. It helps to handle the aws_mqtt_rr_client_operation termination.
+private class StreamingOperationCore: @unchecked Sendable {
     fileprivate var rawValue: OpaquePointer? // <aws_mqtt_rr_client_operation>?
-
-    public init () {
-        // TODO: INIT THE MEMBERS
-        self.rawValue = nil
+    fileprivate let rwlock = ReadWriteLock()
+    fileprivate let options: StreamingOperationOptions
+    
+    fileprivate init (streamOptions: StreamingOperationOptions, client: MqttRequestResponseClientCore) throws {
+        self.options = streamOptions
+        let rawValue = streamOptions.withCPointer(userData: Unmanaged<StreamingOperationCore>.passRetained(self).toOpaque()) { optionsPointer in
+            return aws_mqtt_request_response_client_create_streaming_operation(client.rawValue, optionsPointer)
+        }
+        guard let rawValue else {
+            throw CommonRunTimeError.crtError(CRTError(code: aws_last_error()))
+        }
+        self.rawValue = rawValue
     }
     
     /// Opens a streaming operation by making the appropriate MQTT subscription with the broker.
-    public func open() {
-        // TODO: open the stream
+    fileprivate func open() throws {
+        try rwlock.read {
+            if let rawValue = self.rawValue {
+                if aws_mqtt_rr_client_operation_activate(rawValue) != AWS_OP_SUCCESS {
+                    throw CommonRunTimeError.crtError(CRTError(code: aws_last_error()))
+                }
+            }
+        }
     }
+    
+    /// Closes the operation
+    fileprivate func close() {
+        rwlock.write {
+            aws_mqtt_rr_client_operation_release(self.rawValue)
+            self.rawValue = nil
+        }
+    }
+}
 
+/// The streaming operation, it is automatically closed (and an MQTT unsubscribe triggered) when its destructor is invoked.
+public class StreamingOperation {
+    fileprivate var operationCore: StreamingOperationCore
+    
+    /// Constructor. The end user should only create the operation through MqttRequestResponseClient->createStream()
+    fileprivate init (streamOptions: StreamingOperationOptions, client: MqttRequestResponseClientCore) throws {
+        self.operationCore = try StreamingOperationCore(streamOptions: streamOptions, client: client)
+    }
+    
+    /// Opens a streaming operation by making the appropriate MQTT subscription with the broker.
+    /// - Throws: CommonRunTimeError
+    public func open() throws {
```

Comment:
Are we missing a close() function? opened streams need to be able to be closed to free up streaming subscriptions.

Reply:
Close() is automatically called on destructor. We don't allow end user manually call close() here.